src/Microsoft.AspNet.OData.Shared/TypeHelper.cs

```diff
@@ -456,6 +456,16 @@ internal static IEnumerable<Type> GetLoadedTypes(IWebApiAssembliesResolver assem
             return result;
         }
 
+        internal static Type GetTaskInnerTypeOrSelf(Type type)
+        {
```

Comment:
I think this doesn't cover all cases. See [here](https://github.com/aspnet/Mvc/blob/624a5ed522e8e32a97fb1ba61e25d2302a5abfed/src/Microsoft.AspNetCore.Mvc.Core/Internal/ActionMethodExecutor.cs) (for example, `ValueTask`). Perhaps you could use the [same method](https://github.com/aspnet/Common/blob/57fdf3b2206edda51d7481d036c26fea16578100/shared/Microsoft.Extensions.ObjectMethodExecutor.Sources/CoercedAwaitableInfo.cs#L30) to infer the awaitable type?

Reply:
@aelij Thanks. So far, I think it's enough. 

`ValueTask` struct is added in C# 7.0, it's shipped in "System.Threading.Tasks.Extensions", while this package is not referenced in our products. 

by the way, CoercedAwaitableInfo or AwaitableInfo is internal class.